filetypes.py

```diff
@@ -62,85 +71,101 @@ class PreProcessor(Filetype):
     (?P<val>\d+)            # Capture group for any number of digits
     (?P<unsigned>[uU]?)\)   # Capture group to capture 'U' (zero or one times) and ')'
     """
-    r_var = r'#define\s*([A-Z_]*[VERSION_MAJOR|VERSION_MINOR|VERSION_PATCH])'
 
-    # initializes version based off regex
-    def init_version(self):
+    def get_version_from_file(self):
         content = ""
         # Open file for reading
         with open(self.target_file, 'r', errors='ignore', encoding='utf-8') as input:
             content = input.read()
 
-        # get major
-        matchMaj = re.search(self.r_major, content, re.X)
-        # get minor
-        matchMin = re.search(self.r_minor, content, re.X)
-        # get patch
-        matchPat = re.search(self.r_patch, content, re.X)
+        # Get match objects and save them
+        self._match_obj_major = re.search(self.r_major, content, re.X)
+        self._match_obj_minor = re.search(self.r_minor, content, re.X)
+        self._match_obj_patch = re.search(self.r_patch, content, re.X)
```

Comment:
Is  `self.`  needed here?

Reply:
Yes, unfortunately we have to use `self` when referencing a class attribute from inside a classes method.